channeld/full_channel.c

```diff
@@ -100,15 +101,14 @@ struct channel *new_full_channel(const tal_t *ctx,
 				 u32 lease_expiry,
 				 struct amount_sat funding,
 				 struct amount_msat local_msat,
-				 const struct fee_states *fee_states,
+				 const struct fee_states *fee_states TAKES,
```

Comment:
small question related to my knowledge, what it means this `TAKES` here?

Reply:
It means that the caller can use the take() marker to indicate that the fee_states parameter should be owned by the new_full_channel call.

Normally the caller would have to free it, but if it passes take(fee_states) it must not, since the call will take ownership.

We use this pattern in several places, to avoid copying.

There's another marker, STEALS which means the call will *always* take ownership.